src/librustc/ty/context.rs

```diff
@@ -1462,14 +1462,22 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
         self.cstore.metadata_encoding_version().to_vec()
     }
 
+    /// Get the DefIds of items with MIRs in dependency crates.
+    /// This is for exclusive use by Yorick, as it makes an assumption: that we never want to
+    /// serialise librustc* in the SIR (they are too big and VMs shouldn't use them).
     pub fn metadata_mir_defids(&self) -> DefIdSet {
         let mut res = Vec::new();
         let tbl_lens = self.cstore.def_path_table_lens();
 
         for (cnum, ln) in tbl_lens.iter().enumerate() {
+            let cnum = CrateNum::from_usize(cnum);
+            if self.crate_name(cnum).as_str().starts_with("rustc") {
```

Comment:
Dumb question: should this be `librustc`?

Reply:
No, I tried that originally and had a `dbg!` inside, which didn't execute. The crate name is without the `lib` prefix.

Reply:
Did you leave the `dbg!` in and check that this definitely fires as expected?

Reply:
Yeah I did, to check this was having an effect.

From my scrollback:
```
...
Compiling rustc-main v0.0.0 (/home/vext01/research/ykrustc/src/rustc)                                             
[src/librustc/ty/context.rs:1472] "SKIPPING CRATE" = "SKIPPING CRATE"                                                
[src/librustc/ty/context.rs:1472] self.crate_name(cnum) = rustc_binary                                               
[src/librustc/ty/context.rs:1472] "SKIPPING CRATE" = "SKIPPING CRATE"                                                
[src/librustc/ty/context.rs:1472] self.crate_name(cnum) = rustc_std_workspace_core                                   
[src/librustc/ty/context.rs:1472] "SKIPPING CRATE" = "SKIPPING CRATE"                                                
[src/librustc/ty/context.rs:1472] self.crate_name(cnum) = rustc_std_workspace_alloc                                  
[src/librustc/ty/context.rs:1472] "SKIPPING CRATE" = "SKIPPING CRATE"                                                
[src/librustc/ty/context.rs:1472] self.crate_name(cnum) = rustc_demangle                                             
[src/librustc/ty/context.rs:1472] "SKIPPING CRATE" = "SKIPPING CRATE"                                                
[src/librustc/ty/context.rs:1472] self.crate_name(cnum) = rustc_driver                                               [src/librustc/ty/context.rs:1472] "SKIPPING CRATE" = "SKIPPING CRATE"                                                
[src/librustc/ty/context.rs:1472] self.crate_name(cnum) = rustc_save_analysis                                    
...
```

I'm going to think about this over the weekend...

Reply:
OK.